example_project/measures/BuildResidentialModel/measure.rb

```diff
@@ -108,10 +113,85 @@ def run(model, runner, user_arguments)
+
+      check_dir_exists(resources_dir, runner)
+      check_dir_exists(measures_dir, runner)
+      check_dir_exists(characteristics_dir, runner)
+      check_file_exists(lookup_file, runner)
+      check_file_exists(buildstock_csv_path, runner)
+
+      lookup_csv_data = CSV.open(lookup_file, col_sep: "\t").each.to_a
+
+      # Retrieve all data associated with sample number
+      bldg_data = get_data_for_sample(buildstock_csv_path, args[:building_id], runner)
+
+      # Retrieve order of parameters to run
+      parameters_ordered = get_parameters_ordered_from_options_lookup_tsv(lookup_csv_data, characteristics_dir)
+
+      # Check buildstock.csv has all parameters
+      missings = parameters_ordered - bldg_data.keys
+      if !missings.empty?
+        runner.registerError("Mismatch between buildstock.csv and options_lookup.tsv. Missing parameters: #{missings.join(', ')}.")
+        return false
+      end
+
+      # Check buildstock.csv doesn't have extra parameters
+      extras = bldg_data.keys - parameters_ordered - ['Building', 'sample_weight']
+      if !extras.empty?
+        runner.registerError("Mismatch between buildstock.csv and options_lookup.tsv. Extra parameters: #{extras.join(', ')}.")
+        return false
+      end
+
+      # Retrieve options that have been selected for this building_id
+      parameters_ordered.each do |parameter_name|
+        # Register the option chosen for parameter_name with the runner
+        option_name = bldg_data[parameter_name]
+        register_value(runner, parameter_name, option_name)
+      end
+
+      # Obtain measures and arguments to be called
+      measures = {}
+      parameters_ordered.each do |parameter_name|
+        option_name = bldg_data[parameter_name]
+        print_option_assignment(parameter_name, option_name, runner)
+        options_measure_args, _errors = get_measure_args_from_option_names(lookup_csv_data, [option_name], parameter_name, lookup_file, runner)
+        options_measure_args[option_name].each do |measure_subdir, args_hash|
+          update_args_hash(measures, measure_subdir, args_hash, false)
+        end
+      end      
+
+      resstock_arguments_runner = OpenStudio::Measure::OSRunner.new(OpenStudio::WorkflowJSON.new) # we want only ResStockArguments registered argument values
+      if !apply_measures(measures_dir, { 'ResStockArguments' => measures['ResStockArguments'] }, resstock_arguments_runner, model, true, 'OpenStudio::Measure::ModelMeasure', nil)
+        return false
+      end
+
+      resstock_arguments_runner.result.stepValues.each do |step_value|
+        value = get_value_from_workflow_step_value(step_value)
+        next if value == ''
+        next if step_value.name == 'weather_station_epw_filepath' # we already assign this in the Baseline.rb mapper
+
+        args[step_value.name.to_sym] = value # FIXME: we'd probably want to check that lookup assignments don't conflict with geojson assignments?
```

Comment:
Yes, a lot of this (and code above) is copied from BuildExistingModel. See the PR description for an explanation.